BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m

```diff
@@ -1166,6 +1179,10 @@ - (NSInteger)read:(uint8_t *)buffer
         NSInteger numberOfBytesRead = 0;
 
         numberOfBytesRead = [self.inputStream read:&buffer[totalNumberOfBytesRead] maxLength:(length - (NSUInteger)totalNumberOfBytesRead)];
+        if(self.contentProcessor) {
+            NSData *bodyData = [NSData dataWithBytes:(const void *)(buffer + totalNumberOfBytesRead) length:numberOfBytesRead];
```

Comment:
Consider non-copying constructor.

Reply:
According to the docs(https://developer.apple.com/documentation/foundation/nsdata/1547229-datawithbytesnocopy?language=objc), Non-copying constructor only works on blocks allocated from heap using malloc since it takes ownership of the buffer.